worker/src/queue.rs

```diff
@@ -179,15 +342,343 @@ impl AsRef<JsValue> for Queue {
+    Json,
+    /// Send a String. This content type can be previewed with the List messages from the dashboard feature.
+    Text,
+    /// Send a JavaScript object that cannot be JSON-serialized but is supported by structured clone (for example Date and Map). This content type cannot be previewed from the Cloudflare dashboard and will display as Base64-encoded.
+    V8,
+}
+
+impl Serialize for QueueContentType {
+    fn serialize<S>(&self, serializer: S) -> std::result::Result<S::Ok, S::Error>
+    where
+        S: serde::Serializer,
+    {
+        serializer.serialize_str(match self {
+            Self::Json => "json",
+            Self::Text => "text",
+            Self::V8 => "v8",
+        })
+    }
+}
+
+#[derive(Serialize)]
+#[serde(rename_all = "camelCase")]
+pub struct QueueSendOptions {
+    content_type: Option<QueueContentType>,
+    delay_seconds: Option<u32>,
+}
+
+pub struct MessageBuilder<T> {
+    message: T,
+    delay_seconds: Option<u32>,
+    content_type: QueueContentType,
+}
+
+impl<T: Serialize> MessageBuilder<T> {
+    /// Creates a new message builder. The message must be `serializable`.
+    pub fn message(message: T) -> Self {
+        Self {
+            message,
+            delay_seconds: None,
+            content_type: QueueContentType::Json,
+        }
+    }
+
+    #[must_use]
+    /// The number of seconds to delay a message for within the queue, before it can be delivered to a consumer
+    pub fn with_delay_seconds(mut self, delay_seconds: u32) -> Self {
+        self.delay_seconds = Some(delay_seconds);
+        self
+    }
+
+    #[must_use]
+    /// The content type of the message.
+    /// Default is `QueueContentType::Json`.
+    pub fn with_content_type(mut self, content_type: QueueContentType) -> Self {
+        self.content_type = content_type;
+        self
+    }
+
+    /// Build the message.
+    pub fn build(self) -> SendMessage<T> {
+        SendMessage {
+            message: self.message,
+            options: Some(QueueSendOptions {
+                content_type: Some(self.content_type),
+                delay_seconds: self.delay_seconds,
+            }),
+        }
+    }
+}
+
+pub struct RawMessageBuilder {
+    message: JsValue,
+    delay_seconds: Option<u32>,
+}
+
+impl RawMessageBuilder {
+    /// Creates a new raw message builder. The message must be a `JsValue`.
+    pub fn message(message: JsValue) -> Self {
+        Self {
+            message,
+            delay_seconds: None,
+        }
+    }
+
+    #[must_use]
+    /// The number of seconds to delay a message for within the queue, before it can be delivered to a consumer
+    pub fn with_delay_seconds(mut self, delay_seconds: u32) -> Self {
+        self.delay_seconds = Some(delay_seconds);
+        self
+    }
+
+    /// Build the message with a content type.
+    pub fn build_with_content_type(self, content_type: QueueContentType) -> SendMessage<JsValue> {
```

Comment:
This deviates from `MessageBuilder` because they must specify a `content_type`?

Reply:
Yes, with the MessageBuilder I think it makes sense for it to default to `json` content type. For raw messages I wasn't sure what the default should be, so I made it so the `content_type` must be specified here.

Happy to change it, if you think we should have a default content type for raw messages.